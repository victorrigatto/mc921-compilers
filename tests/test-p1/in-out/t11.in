int f(int m, int n) {
    if (! n < 2) {
        n = n * 2;
        return n;
    }
    else {
        m = m % 2;
        return m;
    }
}

int main() {
    int m = 9;
    int n = 4;
    print(f(m, n));
    return 0;
}
