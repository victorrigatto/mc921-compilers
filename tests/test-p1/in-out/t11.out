LexToken(INT,'int',1,0)
LexToken(ID,'f',1,4)
LexToken(LPAREN,'(',1,5)
LexToken(INT,'int',1,6)
LexToken(ID,'m',1,10)
LexToken(COMMA,',',1,11)
LexToken(INT,'int',1,13)
LexToken(ID,'n',1,17)
LexToken(RPAREN,')',1,18)
LexToken(LBRACE,'{',1,20)
LexToken(IF,'if',2,26)
LexToken(LPAREN,'(',2,29)
LexToken(NOT,'!',2,30)
LexToken(ID,'n',2,32)
LexToken(LT,'<',2,34)
LexToken(INT_CONST,'2',2,36)
LexToken(RPAREN,')',2,37)
LexToken(LBRACE,'{',2,39)
LexToken(ID,'n',3,49)
LexToken(EQUALS,'=',3,51)
LexToken(ID,'n',3,53)
LexToken(TIMES,'*',3,55)
LexToken(INT_CONST,'2',3,57)
LexToken(SEMI,';',3,58)
LexToken(RETURN,'return',4,68)
LexToken(ID,'n',4,75)
LexToken(SEMI,';',4,76)
LexToken(RBRACE,'}',5,82)
LexToken(ELSE,'else',6,88)
LexToken(LBRACE,'{',6,93)
LexToken(ID,'m',7,103)
LexToken(EQUALS,'=',7,105)
LexToken(ID,'m',7,107)
LexToken(MOD,'%',7,109)
LexToken(INT_CONST,'2',7,111)
LexToken(SEMI,';',7,112)
LexToken(RETURN,'return',8,122)
LexToken(ID,'m',8,129)
LexToken(SEMI,';',8,130)
LexToken(RBRACE,'}',9,136)
LexToken(RBRACE,'}',10,138)
LexToken(INT,'int',12,141)
LexToken(ID,'main',12,145)
LexToken(LPAREN,'(',12,149)
LexToken(RPAREN,')',12,150)
LexToken(LBRACE,'{',12,152)
LexToken(INT,'int',13,158)
LexToken(ID,'m',13,162)
LexToken(EQUALS,'=',13,164)
LexToken(INT_CONST,'9',13,166)
LexToken(SEMI,';',13,167)
LexToken(INT,'int',14,173)
LexToken(ID,'n',14,177)
LexToken(EQUALS,'=',14,179)
LexToken(INT_CONST,'4',14,181)
LexToken(SEMI,';',14,182)
LexToken(PRINT,'print',15,188)
LexToken(LPAREN,'(',15,193)
LexToken(ID,'f',15,194)
LexToken(LPAREN,'(',15,195)
LexToken(ID,'m',15,196)
LexToken(COMMA,',',15,197)
LexToken(ID,'n',15,199)
LexToken(RPAREN,')',15,200)
LexToken(RPAREN,')',15,201)
LexToken(SEMI,';',15,202)
LexToken(RETURN,'return',16,208)
LexToken(INT_CONST,'0',16,215)
LexToken(SEMI,';',16,216)
LexToken(RBRACE,'}',17,218)
